src/modules/messagePreview/module.js

```diff
@@ -25,27 +30,43 @@ subscriptions.set('messageCreate', async (client, message) => {
         if (
             !msg.channel
                 .permissionsFor(message.author)
-                ?.has([Permissions.FLAGS.VIEW_CHANNEL, Permissions.FLAGS.READ_MESSAGE_HISTORY])
+                ?.has([
+                    PermissionsBitField.Flags.ViewChannel,
+                    PermissionsBitField.Flags.ReadMessageHistory,
+                ])
         )
             return;
         if (!msg.member) await msg.guild.members.fetch(msg.author);
 
-        const embed = new MessageEmbed()
-            .setAuthor(msg.member.displayName, msg.member.displayAvatarURL({ dynamic: true }))
-            .setDescription(msg.content)
+        const embed = new EmbedBuilder()
+            .setAuthor({
+                name: msg.member.displayName,
+                iconURL: msg.member.displayAvatarURL(),
+                url: msg.url,
+            })
             .setColor(msg.member.displayColor)
-            .setFooter(`Requested by ${message.member.displayName}`)
-            .setTimestamp(msg.createdTimestamp);
+            .setFooter({ text: `Requested by ${message.member.displayName}` })
+            .setTimestamp(msg.createdTimestamp)
+            .setURL(msg.url);
```

Comment:
this is set above already

Reply:
The above sets the author. This sets the title.

Reply:
it clearly sets url: msg.url
and here you do .setURL(msg.url)

Reply:
Oh my bad.